### PR TITLE
docs: consistently name Consul service mesh

### PR DESCRIPTION
**Pull request by jkirschner-hashicorp**
Remove outdated usage of "Consul Connect" instead of Consul service mesh.

The connect subsystem in Consul provides Consul's service mesh capabilities. However, the term "Consul Connect" should not be used as an alternative to the name "Consul service mesh".

### Post PR Follow-Up

- [ ] Go through the Consul tutorials and make a similar update.
- [ ] Modify consul CLI help text that uses "connect" where appropriate

### CODE REVIEW

File: website/content/commands/connect/ca.mdx [hunk 1]
```diff
@@ -34,8 +35,8 @@ Usage: consul connect ca <subcommand> [options] [args]
   For more examples, ask for subcommand help or view the documentation.
 
 Subcommands:
-    get-config    Display the current Connect Certificate Authority (CA) configuration
-    set-config    Modify the current Connect CA configuration
+    get-config    Display the current service mesh Certificate Authority (CA) configuration
```
Comment (jkirschner-hashicorp) [inline website/content/commands/connect/ca.mdx:38]: Note: I intend to submit a separate PR to change the CLI command help descriptions output by the code itself

File: website/content/docs/connect/proxies/envoy.mdx [hunk 2]
```diff
@@ -7,7 +7,7 @@ description: >-
 
 # Envoy Proxy Configuration for Service Mesh
 
-Consul Connect has first class support for using
+Consul servic emesh has first class support for using
```
Comment (im2nguyen) [inline website/content/docs/connect/proxies/envoy.mdx:10]:
```suggestion
Consul service mesh has first class support for using
```

File: website/content/docs/connect/proxies/envoy.mdx [hunk 3]
```diff
@@ -79,11 +79,11 @@ The dynamic configuration Consul Connect provides to each Envoy instance include
 - Configuration to [expose specific HTTP paths](/consul/docs/connect/registration/service-registration#expose-paths-configuration-reference).
 
 For more information on the parts of the Envoy proxy runtime configuration
-that are currently controllable via Consul Connect see [Dynamic
+that are currently controllable via Consul service mesh, refer to [Dynamic
 Configuration](#dynamic-configuration).
 
 We plan to enable more and more of Envoy's features through
-Connect's first-class configuration over time, however some advanced users will
+Consul's first-class configuration over time, however some advanced users will
```
Comment (im2nguyen) [inline website/content/docs/connect/proxies/envoy.mdx:86]:
```suggestion
Consul service mesh's first-class configuration over time, however some advanced users will
```

File: website/content/docs/connect/proxies/envoy.mdx [hunk 4]
```diff
@@ -287,11 +287,11 @@ the [global `proxy-defaults` configuration
 entry](/consul/docs/connect/config-entries/proxy-defaults) to act as
 defaults that are inherited by all services.
 
-- `protocol` - The protocol the service speaks. Connect's Envoy integration
+- `protocol` - The protocol the service speaks. Consul's Envoy integration
```
Comment (im2nguyen) [inline website/content/docs/connect/proxies/envoy.mdx:290]:
```suggestion
- `protocol` - The protocol the service speaks. Consul service mesh's Envoy integration
```

File: website/content/docs/connect/proxies/envoy.mdx (hunk 3 above)
Comment (im2nguyen) [reply website/content/docs/connect/proxies/envoy.mdx:86]: not sure if we need to be explicit here, feel free to ignore

File: website/content/docs/connect/proxies/envoy.mdx [hunk 5]
```diff
@@ -648,7 +648,7 @@ definition](/consul/docs/connect/registration/service-registration) or
 - `envoy_extra_static_listeners_json` - Similar to
   `envoy_extra_static_clusters_json` but appends one or more [Envoy listeners](https://www.envoyproxy.io/docs/envoy/v1.17.2/api-v3/config/listener/v3/listener.proto) to the array of [static
   listener](https://www.envoyproxy.io/docs/envoy/v1.17.2/api-v3/config/bootstrap/v3/bootstrap.proto#envoy-v3-api-field-config-bootstrap-v3-bootstrap-staticresources-listeners) definitions.
-  Can be used to setup limited access that bypasses Connect mTLS or
+  Can be used to setup limited access that bypasses service mesh mTLS or
```
Comment (im2nguyen) [inline website/content/docs/connect/proxies/envoy.mdx:651]:
```suggestion
  Can be used to setup limited access that bypasses the service mesh's mTLS or
```

**Review by david-yu — APPROVED** [review]

File: website/content/docs/upgrading/upgrade-specific.mdx [hunk 6]
```diff
@@ -851,13 +851,13 @@ as soon as possible after upgrade, as well as updating any integrations to work
...
-### Connect Multi-datacenter
+### Multi-datacenter service mesh
 
 This only applies to users upgrading from an older version of Consul Enterprise to Consul Enterprise 1.4.0 (all license types).
 
-In addition, this upgrade will only affect clusters where [Connect is enabled](/consul/docs/connect/configuration) on your servers before the migration.
+In addition, this upgrade will only affect clusters where [service mesh is enabled](/consul/docs/connect/configuration) on your servers before the migration.
 
-Connect multi-datacenter uses the same primary/secondary approach as ACLs and
+Multi-datacenter service-mesh uses the same primary/secondary approach as ACLs and
```
Comment (im2nguyen) [inline website/content/docs/upgrading/upgrade-specific.mdx:860]:
```suggestion
Multi-datacenter service mesh uses the same primary/secondary approach as ACLs and
```

**Review by im2nguyen — APPROVED** [review]
overall looks great, thank you so much for doing this!!

File: website/content/commands/connect/ca.mdx (hunk 1 above)
Comment (jkirschner-hashicorp) [reply website/content/commands/connect/ca.mdx:38]: CLI help text update PR submitted for review by eng: https://github.com/hashicorp/consul/pull/17230
